application-admintools-default/src/main/java/com/xwiki/admintools/internal/health/checks/MemoryHealthCheck.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.admintools.internal.health.checks;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.configuration.ConfigurationSource;
+import org.xwiki.localization.ContextualLocalizationManager;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xwiki.admintools.health.HealthCheck;
+import com.xwiki.admintools.health.HealthCheckResult;
+
+/**
+ * Implementation of {@link HealthCheck} for checking instance memory performance.
+ *
+ * @version $Id$
+ */
+@Component
+@Named(MemoryHealthCheck.HINT)
+@Singleton
+public class MemoryHealthCheck implements HealthCheck
+{
+    /**
+     * Component identifier.
+     */
+    public static final String HINT = "CACHE_HEALTH_CHECK";
+
+    @Inject
+    protected ContextualLocalizationManager localization;
+
+    @Inject
+    protected Provider<XWikiContext> xcontextProvider;
+
+    @Inject
+    @Named("xwikicfg")
+    private ConfigurationSource configurationSource;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public HealthCheckResult check()
+    {
+        boolean cacheOk = isEnoughCache();
+        boolean memoryOk = hasEnoughMemory();
+        if (cacheOk && memoryOk) {
+            return new HealthCheckResult();
+        }
+        return new HealthCheckResult("There are memory issues!", "dummy memory help link");
```

Comment:
This was one subject that I wanted to talk with you, I forgot to add the comment. After discussing with Lavinia and Teodor, I decided with Lavinia that the best course of action would be to use the logger instead of the HealthCheckResult and show the status of the health check inside the job logs (as you can see in the image above) and create a dedicated page that has some useful links to for various issues. Because of this, the HealthCheckResult has no purpose anymore. Should I keep this class for a future use, or simply remove it and adapt the code?

Reply:
> I decided with Lavinia that the best course of action would be to use the logger instead of the HealthCheckResult and show the status of the health check inside the job logs

My problem with this approach is that the health check job log can contain messages (errors, warnings) that are not related to the health check. Any API that the health check job calls directly or indirectly can log its own messages which raises the question: how do you know which log messages are health check errors / warnings and which are not. Or otherwise put, how do you know that there aren't any health check errors / warnings in the log?